pinot-core/src/main/java/org/apache/pinot/core/data/table/IndexedTable.java

```diff
@@ -84,7 +93,45 @@ public boolean upsert(Record record) {
     return upsert(new Key(keyValues), record);
   }
 
-  public abstract int getNumResizes();
+  @Override
+  public int size() {
+    return _topRecords != null ? _topRecords.size() : _lookupMap.size();
+  }
+
+  /**
+   * Resizes the lookup map based on the trim size.
+   */
+  protected void resize() {
+    long startTimeMs = System.currentTimeMillis();
+    _tableResizer.resizeRecordsMap(_lookupMap, _trimSize);
+    long resizeTimeMs = System.currentTimeMillis() - startTimeMs;
```

Comment:
This clock is subject to adjustments, so can jump forwards and backwards. For relative time measurement of synchronous code, System.nanoTime() is the right clock to use.

Reply:
Good point. Changed to use ns to track the resize time